bin/taskcat

```diff
@@ -57,11 +59,21 @@ def main():
                     tcat_instance.set_config("%s/%s" % (taskcat_cfg['global']['qsname'], config_path))
             except Exception as e:
                 print(taskcat.PrintMsg.ERROR + str(e))
+            project_path = '/'.join(tcat_instance.get_config().split('/')[0:-3])
+            project_name = '/'.join(tcat_instance.get_config().split('/')[-3:-2])
```

Comment:
These two lines confuse me. IIRC the taskcat config is loaded as dictionary into the stack. We have a 'qsname' key in that config.Why are we doing a `.join()` plus a `.split()`, plus operations on an index. Am I missing something here? We

Reply:
I agree that the intent here is not clear, we should probably be setting up the project path related items in a better way. Perhaps a few util functions with clear names would help.

Up till recently (a pr or two ago) you could only launch tcat from the parent of the project folder. This allows you to execute taskcat from anywhere relative to the project.

```
jmmccon@8c85901f90a3 /p/tmp> taskcat -c ~/PycharmProjects/aws-sb/quickstart-redhat-openshift/ci/taskcat.jmmccon-inner.yml  -l strict

...

[ERROR  ] :Exiting due to lint warnings
```

there is no straightforward way(that I can see) to tell where your `pwd` is in relation to the project root. 

`project_path` assumes that the ci config is in the location `project_dir/ci/config.yaml`, so using the example above:
```
In [2]: config="~/PycharmProjects/aws-sb/quickstart-redhat-openshift/ci/taskcat.jmmccon-inner.yml"

In [4]: config.split('/')
Out[4]: ['~', 'PycharmProjects', 'aws-sb', 'quickstart-redhat-openshift', 'ci', 'taskcat.jmmccon-inner.yml']

In [5]: config.split('/')[0:-3]
Out[5]: ['~', 'PycharmProjects', 'aws-sb']

In [6]: "/".join(config.split('/')[0:-3])
Out[6]: '~/PycharmProjects/aws-sb'
```
So that gives us the parent path to the project path, similarly `project_name` gives us the name of the directory containing the project.

Reply:
@agamzn I don't believe this should be a blocker for this pr. Especially considering all that I'm doing is refactoring to (arguably) improving the readability of logic already present https://github.com/aws-quickstart/taskcat/blob/develop/bin/taskcat#L61-L64

Reply:
@jaymccon I'll concede that. The rejection was to ensure we clarified the intent here prior to a merge. It wasn't explicitly meant as a blocker to the PR. I'm happy to resolve the objection. 👍